src/components/LoginButton/index.tsx

```diff
@@ -0,0 +1,18 @@
+import { useAuthContext } from "../../store/auth/AuthContext";
+import "./loginButton.css";
+
+// const clientId = "817921738258-jfbapkf5tsqmbgjrn672ua2udsuta7vt.apps.googleusercontent.com";
+
+const LoginButton = () => {
```

Comment:
> Nice work, well done! 👏 Few questions from my side though.
> 
> Will you be working on the logout feature in a separate PR?
> 
> As for the tests, we should test the behavior of the Authentication context, since it contains the main logic. I'd say that the most important use case here is to check whether the logged in user will be able to access the homepage and books or not. To be able to test the behavior of the Authentication context, we should mock it first. Those tests could look like this:
> 
> ```
> import React from "react";
> import { render, screen } from "@testing-library/react";
> 
> import App from "../../App";
> import * as AuthContext from "./AuthContext";
> 
> jest.spyOn(AuthContext, "useAuthContext").mockImplementation();
> 
> describe("Test authentication context", () => {
>   test("should display the home page with books if the user is logged in", async () => {
>     jest.spyOn(AuthContext, "useAuthContext").mockImplementation(() => ({
>       isLoggedIn: true,
>       loaded: true,
>       userProfile: {
>         name: "test",
>         imageUrl: "http://test.com",
>         email: "test@test.com",
>       },
>       redirected: true,
>     }));
> 
>     render(<App />);
> 
>     expect(screen.getAllByText("Book title")).toHaveLength(4);
>     expect(screen.getAllByText("Book2")).toHaveLength(4);
>   });
> 
>   test("should not display the home page with books if the user is not logged in", async () => {
>     jest.spyOn(AuthContext, "useAuthContext").mockImplementation(() => ({
>       isLoggedIn: false,
>       loaded: true,
>       userProfile: {
>         name: "test",
>         imageUrl: "http://test.com",
>         email: "test@test.com",
>       },
>       redirected: true,
>     }));
> 
>     render(<App />);
> 
>     expect(screen.queryAllByText("Book title")).not.toHaveLength(4);
>     expect(screen.queryAllByText("Book2")).not.toHaveLength(4);
>   });
> });
> ```
> 
> You can further test the behavior of the Authentication context if you like 🙂
> 
> Since you use `reload` function, I believe you will have the following error in the test logs => `Error: Not implemented: navigation (except hash changes)`. [This Jest plugin](https://npmmirror.com/package/jest-location-mock) could fix that error.
> 
> Please let me know if case of any questions.

- We decided that logout will be implemented on separate branch.
- I figured out that reload was redundant, so I removed it.